<?xml version="1.0"?>
<launch>
  <arg name="model" default="burger" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="x_pos" default="0.0"/>
  <arg name="y_pos" default="0.0"/>
  <arg name="z_pos" default="0.0"/>
  <arg name="gui" default="false"/>

  <!-- We need to set this environment variable because they messed up in urdfdom
       See this: https://github.com/ros-visualization/rviz/issues/1249 -->
  <env name="LC_NUMERIC" value="en_US.UTF-8"/>

  <!-- Set robot pose -->
  <node pkg="rosservice" type="rosservice" name="set_robot_pose" output="screen" args="call /gazebo/set_model_state '{model_state: { model_name: turtlebot3, pose: { position: { x: $(arg x_pos), y: $(arg y_pos), z: $(arg z_pos) }, orientation: {x: 0, y: 0, z: 0, w: 1 } }, twist: { linear: {x: 0 , y: 0, z: 0 } , angular: { x: 0, y: 0, z: 0 } } , reference_frame: map } }'" />

  <!-- Stop the robot -->
  <node pkg="rostopic" type="rostopic" name="stop_robot_pub" args="pub --once /cmd_vel geometry_msgs/Twist '{linear: { x: 0.0, y: 0.0, z: 0.0 }, angular: { x: 0.0, y: 0.0, z: 0.0 } }'" />

  <!-- Reset SLAM -->
  <node pkg="rostopic" type="rostopic" name="reset_slam_pub" args="pub --once /syscommand std_msgs/String '{data: reset }'" />

  <!-- Exploration -->
  <node pkg="irob_assignment_1" type="explorer.py" name="explorer" output="screen" />

  <!-- Collision avoidance -->
  <node pkg="irob_assignment_1" type="collision_avoidance.py" name="collision_avoidance" output="screen" />

  <!-- Controller -->
  <node pkg="irob_assignment_1" type="controller.py" name="controller" output="screen" />

  <!-- Costmap-->
  <node name="costmap_node" pkg="costmap_2d" type="costmap_2d_node">
    <rosparam file="$(find irob_assignment_1)/config/costmap.yaml" command="load" ns="costmap" />
    <remap from="base_scan" to="scan"/>
  </node>

</launch>
